bootloader/src/pyi_utils.c

```diff
@@ -789,6 +802,7 @@ pyi_utils_create_child(const char *thisfile, const int argc, char *const argv[])
     /* Child code. */
     if (pid == 0) {
         /* Replace process by starting a new application. */
+        set_systemd_env(getpid());
```

Comment:
I'd move the `getpid()` call into the `if` within `set_systemd_env`. The the resp. env-var is not set, nothing needs to be done at all.